processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -511,6 +538,76 @@ private void applyParameterNameBasedMapping() {
             }
         }
 
+
+        private ConstructorMapping createConstructorMapping() {
+            Constructor constructor = method.getResultType().getConstructor();
+
+            if (constructor != null) {
+                LinkedHashMap<String, PropertyMapping> mappings = new LinkedHashMap<String, PropertyMapping>(  );
+
+                for (String propertyName : constructor.getPropertyNames()) {
+                    if ( constructorPropertyMappings.containsKey( propertyName ) ) {
+                        mappings.put( propertyName, constructorPropertyMappings.get( propertyName ) );
+                    }
+                    else {
+                        Parameter parameter = constructor.getParameter( propertyName );
```

Comment:
Replace missing mappings with either a null assignment or the relevant primitive constant